airflow/kubernetes/kube_config.py

```diff
@@ -45,7 +45,10 @@ def __init__(self):
 
         self.worker_container_repository = conf.get(self.kubernetes_section, 'worker_container_repository')
         self.worker_container_tag = conf.get(self.kubernetes_section, 'worker_container_tag')
-        self.kube_image = f'{self.worker_container_repository}:{self.worker_container_tag}'
+        if self.worker_container_repository and self.worker_container_tag:
```

Comment:
we could probably loosen this to allow user to specify image only (and not tag) but this is the current behavior.  currently if either is not specified it won't work unless specified in executor_config